기본 폴더 및 파일 생성

mjjeon2645 opened the pull request:
기본 폴더(images, css 폴더) 및 인덱스 파일 생성

mjjeon2645 committed:
기본 폴더 및 파일 생성

기본 폴더(images, css 폴더) 및 인덱스 파일 생성

mjjeon2645 committed:
project 영역까지 html 1차 작성

project 영역까지 html 1차 작성

mjjeon2645 committed:
빈 줄 추가

마지막 빈 줄 추가

mjjeon2645 committed:
html 1차 작성 완료

html 1차 작성 완료 및 css 파일 추가 예정

mjjeon2645 committed:
인덱스 전역에 지정할 css 파일 생성

인덱스 html 내용 전반에 지정할 css 파일 생성함

mjjeon2645 committed:
헤더와 나브 구현

현재 창을 늘렸을 때 레이아웃이 맞지 않고 컨택트미 버튼이 높낮이가 다른 부분이 있어 추후 수정 필요

mjjeon2645 committed:
상단 헤더 수정

상단 헤더 비뚤어짐 수정함

mjjeon2645 committed:
프로필 중 사진, 본인정보란 구현

사진, 본인정보란 구현

mjjeon2645 committed:
프로젝트 영역 1차 구현

그림자가 뜨면서 레이아웃에 약간 변형이 생기는 부분 추후 수정 필요

mjjeon2645 committed:
experience 섹션 구현

experience 섹션 구현

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,213 @@
+                                <p class="job">Product Developer</p>
+                            </div>
+                            <p class="mention">Be curious, not judgemental</p>
+                            <div class="tags">
+                                <span>나의 관심 분야 태그</span>
+                                <span>나의 관심 분야 태그</span>
+                                <span>내가 할 수 있는 개발언어</span>
+                            </div>
```
span은 무언가 꾸며주기 위함일 뿐 아무런 의미가 없습니다. 여기서는 태그들이라는 의미가 있는데 div에 span으로 넣어주기 보다는 태그 목록들이니 ul, li 태그를 사용하는 것이 더 적절해 보입니다.

mjjeon2645 replied:
네~ 리팩터링 해보겠습니다.

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,213 @@
+                            </div>
+                        </div>
+                        <div class="contacts">
+                            <p>TEL.</p>
+                            <p>(+82) 01056341740</p>
+                            <p>Email.</p>
+                            <p>mjjeon2645@gmail.com</p>
+                            <p>Blog.</p>
+                            <a href="https://velog.io/@shanti">https://velog.io/@shanti</a>
+                            <p>GitHub.</p>
+                            <a href="https://github.com/mjjeon2645">https://github.com/mjjeon2645</a>
```
종목과 내용의 한 쌍이니 더 의미있는 태그를 사용할 수 있습니다. 
dl, dt, dd 에 대해 학습하시면 또 이런 쌍을 이루는 마크업 작업 하실 때 어떤 태그를 써야하는 지 빠르게 감이 오실거에요 :)

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dt

mjjeon2645 replied:
넵 사실 li를 사용하려고 하니 너무 맥락에 맞지않는것 같아서 고민했는데 답은 다른 곳에 숨어있었군요ㅎㅎ

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,213 @@
+                    <ol>
+                        <li>
+                            <div class="project-container">
+                                <div class="project-image">img</div>
```
img라는 실제 문구를 넣는 것은 이 내용에 의미가 드러나지 않습니다. 
의미가 없는경우 img라는 말보다 아예 없는 것이 더 나을 수도 있어요.
흔히 이미지에 텍스트를 꼭 넣어주는 이유는 시각장애를 가지신 분들이 스크린 리더기로 정보를 쉽게 얻을 수 있게 하기 위해서 입니다. 
그러나 의미없는 텍스트가 들어간다면, 듣는 사람으로 하여금 불필요한 내용까지 듣게 되어 굉장히 피로해집니다. 그래서 요즘에는 이미지에 alt 속성을 적지 않는 이유 추세입니다. 

항상 스크린 리더로 실제로도 들어보시고 들었을 때 어떻게 나올지도 예상해 보면서 작업해주세요 :)

mjjeon2645 replied:
ㅇ ㅏ... alt 속성도 적지 않는 추세이군요??? 처음 알았습니다.
스크린 리더를 저희가 사용할 수 있다는 점도 몰랐네요.
우선 디자인 상으로만 이미지라고 표현되어있어 문구를 써놓은건데 다른걸로 대체해볼까 싶네요. 더 신경써보겠습니다~

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,213 @@
+                            <div class="project-container">
+                                <div class="project-image">img</div>
+                                    <div class="project-text">
+                                        <p class="project-title">프로젝트에 관한 설명</p>
+                                        <p class="project-detail">스킬에 관한 설명이 들어가는 영역입니다. <br> 
```
이미 프로젝츠 안에 존재하는 내용들입니다. projcet-title, project-detail과 같이 클래스 네임에서 계층까지 모두 알려주기 보다는 간결하고 정확한 클래스네임을 지어서 CSS  selector를 알맞게 쓰시는 것이 훨씬 좋습니다. 

예를 들어 이 title은 #projects .title 이 되겠죠?

mjjeon2645 replied:
네네. 클래스네임을 일일이 다 디테일하게 쓰다보니 이렇게까지 하는게 오히려 피로감을 줄수 있다는 생각이 들더라구요. 수정해보겠습니다~

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,213 @@
+                        <h2>Experience</h2>
+                        <span>경력 및 학력이 들어가는 영역입니다</span>
+                    </div>
+                    <div class="row">
```
row라는 클래스 네임은 정말 아무 의미가 없어요 ㅜㅜ 
여기 있는 모든 요소들을 다 row라고 부를 수 있습니다. 의미없는 이름을 지어주기 보단 클래스 네임이 없는게 더 깔끔할 수 있습니다.

mjjeon2645 replied:
넵 알겠습니다~!

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,213 @@
+                    <div class="experience-title">
+                        <h2>Experience</h2>
+                        <span>경력 및 학력이 들어가는 영역입니다</span>
+                    </div>
+                    <div class="row">
+                        <span class="title">회사 1</span>
+                        <span class="detail">주요업무를 적는란</span>
+                        <span class="period">2022.12.00~2022.12.00</span>
+                    </div>
+                    <div class="row">
+                        <span class="title">회사 2</span>
+                        <span class="detail">주요업무를 적는란</span>
+                        <span class="period">2022.12.00~2022.12.00</span>
+                    </div>
+                    <div class="row">
+                        <span class="title">학교 1 및 교육기관</span>
+                        <span class="detail">성수 코딩 도장</span>
+                        <span class="period">2022.12.00~2022.12.00</span>
+                    </div>
+                    <div class="row">
+                        <span class="title">학교 1 및 교육기관</span>
+                        <span class="detail">이태원 코딩 도장</span>
+                        <span class="period">2022.12.00~2022.12.00</span>
+                    </div>
```
비슷한 내용이 목록으로 나오고 있네요. 
그러면 div보다는 리스트 태그들을 이용해주는게 더 의미가 있습니다. 
그렇게 하면 row라는 이름없이도 그 의미가 저절로 포함 될 수 있습니다.

mjjeon2645 replied:
넵. 의미없는 클래스네임+div의 조합보다 의미있는 태그로 묶어주는 것. 연습할게요~

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,213 @@
+                <section id="retrospect">
+                    <h2>Retrospect</h2>
+                    <p>회고 모음이 여기에 들어갑니다</p>
+                    <ul class="retrospect-category">
+                        <li>
+                            <a href="#">분류 1</a>
+                        </li>
+                        <li>
+                            <a href="#">분류 2</a>
+                        </li>
+                        <li>
+                            <a href="#">분류 3</a>
+                        </li>
+                    </ul>
```
디자인엔 있지만 지금 우리는 분류를 해줄 수 없으니 이 부분은 생략해주셔도 좋습니다 😂

mjjeon2645 replied:
아 li를 적용하지 않는 방향으로 수정하겠습니다 ^^

bbhye1 commented on css/experience.css:
```diff
@@ -0,0 +1,42 @@
+}
+
+.experience-title h2 {
+    font-size: 2em;
+    display: inline-block;
+    font-family: 'Lato', sans-serif; 
+    font-weight: 700;
```
font 관련된 속성은 함께 묶어주세요. 
display를 마지막에 넣어주세요 ;)

mjjeon2645 replied:
넵 앞으로 작성할 때 해당 가이드 따라 진행하겠습니다 :)

bbhye1 commented on css/experience.css:
```diff
@@ -0,0 +1,42 @@
+}
+
+.experience-title span {
+    padding-left: .5em;
+    color: #666666;
+    font-family: 'Spoqa Han Sans', 'Spoqa Han Sans JP', 'Sans-serif';
+    font-size: .8em;
```
CSS 속성은 box-model을 따라서 작성해주세요. 
font-size 속성은 이 이후 em으로 잡아주는 모든 것에 영향을 끼칩니다. 지금 이 코드에서도 padding과 font-size 속성을 바꾸면 스타일이 바뀔거에요.  항상 font관련 속성은 제일 위에 작성해주세요.

mjjeon2645 replied:
폰트는 가장 위에.... 기억하겠습니다!

bbhye1 commented on css/experience.css:
```diff
@@ -0,0 +1,42 @@
+}
+
+.row {
+   height: 5.5vh;
+   border-bottom: 1px dashed #DDDDDD;
+   padding-top: 2.2vh;
```
최대한 vh와 em을 섞어 쓰지 않도록 하는 것이 좋습니다. 
단위는 파악하기 쉽게 통일해주셔야 합니다. 여기저기서 다른 단위를 사용하면 읽는 사람에게 혼동을 줄 수 있어요. 
대부분 Desktop 기준은 em을, 기기가 훨씬 더 다양하게 고정되어 있는 모바일의 경우에는 vw단위로 사용합니다.

mjjeon2645 replied:
아... 데탑은 em, 모바일은 vw 참고하겠습니다.
지금은 데탑 기준으로 작성중이니 em을 활용하면 되겠네요~!

bbhye1 commented on css/header.css:
```diff
@@ -0,0 +1,64 @@
+}
+
+header .contact {
+    font-family: 'Lato', sans-serif;
+    font-weight: 500;
+    padding-inline: 20px;
+    background-color: #000;
+    border-radius: 30px;
+    padding-top: 5px;
+    padding-bottom: 5px;
```
여기도 관련있는 속성끼리 묶어주면 코드를 빠르게 이해하기 쉽겠죠?

mjjeon2645 replied:
폰트는 묶어서 가장 위로, 그다음 패딩요소를 묶었고 나머지를 아래에 위치시켜보았습니다~

bbhye1 commented on css/profile.css:
```diff
@@ -0,0 +1,98 @@
+    display: grid;
+    grid-template-columns: 1fr 4fr;
+    grid-template-rows: 1fr 1fr;
+    grid-template-areas: "main-photo detail"
+    "main-photo contacts"
+    ;
```
indent를 맞춰주세요. 
코드에서도 깔끔하게 정리되어 있어야 이 코드를 읽는 다른 개발자가 빠르고 쉽게 이해할 수 있습니다~ :)

mjjeon2645 replied:
넵~!

mjjeon2645 committed:
임시저장

임시저장

mjjeon2645 committed:
회고란 구현

회고섹션 구현. 창을 늘이고 줄일 때마다 다크섹션이 깨지는 부분 추후에 베스트 프랙티스 보고 수정할 것

mjjeon2645 committed:
마무리 문구 섹션 구현

마무리 문구 섹션 구현

mjjeon2645 committed:
푸터 및 맨위로 버튼 구현

푸터, 맨 위로 가는 버튼 구현

mjjeon2645 committed:
리뷰 반영하여 1차 리팩터링

리뷰 반영하여 1차 리팩터링.
2차 리팩터링은 미디어쿼리 구현한 후에 진행할 예정

mjjeon2645 committed:
베스트프랙티스 참고하여 html 및 css 전면수정

모바일버전 구현 시 기존의 html 작성이 미비하여 구현이 불가능한 상황
따라서 베스트프랙티스를 참고하여 html과 css를 함께 수정
현재 상단 헤더부분에 대한 구현 완료

mjjeon2645 committed:
상단 프로필까지 구현 완료

상단 프로필 모바일내용까지 구현 완료

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,224 @@
+        </header>
+        <main>
+            <article id="introduce">
+                <div class="photo">
```
사진이 의미하는 바는 굉장히 광범위하게 느껴져요. img에 들어가는 것은 대부분 사진이죠? 
이렇게 프로필 이미지를 넣는 곳은 통상적으로 avatar라고 부릅니다.

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,224 @@
+                <section class="address">
+                    <ul>
+                        <li>
+                            <address>
+                                <dt>TEL.</dt>
+                                <dd>(+82) 01056341740</dd>
+                            </address>
```
dt와 dd를 감싸는 태그는 dl입니다. 
https://developer.mozilla.org/ko/docs/Web/HTML/Element/dl

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,224 @@
+                                <div class="project-image"></div>
+                                    <div class="project-text">
+                                        <p class="project-title">프로젝트에 관한 설명</p>
+                                        <p class="project-detail">스킬에 관한 설명이 들어가는 영역입니다. <br> 
```
br은 사용해서는 안되는 태그입니다. 
전혀 시멘틱하지 않은 마크업의 대명사죠ㅎㅎ 
br을 쓰기 보다 문장을 span으로 감싸주어 display를 block으로 만들어 줄바꿈을 해줄 수도 있고 
제목에서 썼던 것처럼 문장 중 강조하고 싶은 부분을 strong 혹은 덜 중요한 부분을 small로 감싸주어 똑같은 효과를 낼 수 있습니다. 
여기서 더 추천드리는 방법은 후자의 방법입니다.

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,224 @@
+                                            최대 두 줄이 들어갑니다
+                                        </p>
+                                        <button class="project-more" type="button">
+                                            <span>버튼을 눌러라</span>
+                                            <img src="/images/chevron-right.png">
```
여기서 span으로 감싸주지 않아도 똑같은 스타일을 button에 적용시킬 수 있을 거에요~ 
span은 꾸미기용의 의미밖에 없는 태그라 다른 방법으로 같은 효과를 낼 수 있다면 사용하지 않는 것이 더 좋습니다. :)

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,224 @@
+                        </li>
+                        <li>
+                            <div class="project-container">
+                                <div class="project-image"></div>
+                                    <div class="project-text">
```
같은 depth의 태그들인 것 같아 보입니다. 
태그들의 depth 도 잘 정렬해주세요~ ;)

mjjeon2645 committed:
프로젝트 부분 html 전면수정 및 모바일 구현

프로젝트 부분 html 전면 수정 및 모바일 구현

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,224 @@
+                        <h2>Experience</h2>
+                        <span>경력 및 학력이 들어가는 영역입니다</span>
+                    </div>
+                    <ul>
+                        <li class="title">회사 1</li>
+                        <li class="detail">주요업무를 적는란</li>
+                        <li class="period">2022.12.00~2022.12.00</li>
+                    </ul>
+                    <ul>
+                        <li class="title">회사 2</li>
+                        <li class="detail">주요업무를 적는란</li>
+                        <li class="period">2022.12.00~2022.12.00</li>
+                    </ul>
+                    <ul>
+                        <li class="title">학교 1 및 교육기관</li>
+                        <li class="detail">성수 코딩 도장</li>
+                        <li class="period">2022.12.00~2022.12.00</li>
+                    </ul>
+                    <ul>
+                        <li class="title">학교 1 및 교육기관</li>
+                        <li class="detail">이태원 코딩 도장</li>
+                        <li class="period">2022.12.00~2022.12.00</li>
+                    </ul>
+                </section>
```
ul들이 나열되어 있는 걸 보니 ul이 li가 되어야 하는 것처럼 보입니다ㅎㅎ

bbhye1 commented on index.html:
```diff
@@ -0,0 +1,225 @@
+                            <a href="#">분류 3</a>
+                        </div>
+                    </div>
+                    <div class="dark-section"></div>
```
이 부분은 뒷 배경 컬러를 의미하는 것 같아 보이네요. 
여기가 아닌 section에 스타일을 주어 같은 효과를 낼 수 있을 거에요. 
불필요한 태그는 좋지 않습니다. ㅎㅎ

bbhye1 commented on css/ending.css:
```diff
@@ -0,0 +1,38 @@
+}
+
+#ending a {
+    color: #FFF;
+    font-family: 'Lato', sans-serif;
```
color는 레이아웃에 영향을 주는 요소가 아닌 꾸며주는 요소라 마지막에 위치시켜주는 것이 좋습니다 :)

bbhye1 commented on css/experience.css:
```diff
@@ -0,0 +1,48 @@
+}
+
+.experience-title span {
+    font-size: .8em;
+    padding-left: .5em;
+    color: #666666;
+    font-family: 'Spoqa Han Sans', 'Spoqa Han Sans JP', 'Sans-serif';
```
font관련 속성은 함께 묶어주세요~

bbhye1 commented on css/experience.css:
```diff
@@ -0,0 +1,48 @@
+}
+
+#experiences ul {
+   height: 5.5vh;
+   border-bottom: 1px dashed #DDDDDD;
+   padding-top: 2.2vh;
```
box-model 순서대로 하면 padding -> height -> border가 되겠죠?

bbhye1 commented on css/go-top.css:
```diff
@@ -0,0 +1,18 @@
+    background: url(/images/top-white.png) no-repeat 0 50%;
+    background-size: contain;
+    position: fixed;
+}
```
마지막 빈줄은 언제나 신경써주세요~

bbhye1 commented on css/go-top.css:
```diff
@@ -0,0 +1,18 @@
+    overflow: hidden;
+    background: url(/images/top-black.png) no-repeat 0 50%;
+    background-size: contain;
+    position: fixed;
+    bottom: 10%;
+    right: 10%;
```
position에 관련된 스타일들은 font 다음 최상단에 위치시켜주세요~ layout에 직접적으로 영향을 미치는 속성들입니다.

mjjeon2645 committed:
경력 및 학력 영역 수정

경력 및 학력 영역 html 전면 수정 및 모바일-데스크탑 디자인 적용

mjjeon2645 committed:
회고영역 완료

회고영역 구현 완료

mjjeon2645 committed:
프로필 1차 완성

프로필 1차 완성

mjjeon2645 committed:
파일 경로 수정

파일 경로 수정

mjjeon2645 committed:
수정본 추가 커밋

수정본 추가 커밋
